### ISSUE TITLE
invalid skaffold.yaml causes skaffold to crash

### ISSUE THREAD

**Issue by msonnleitner** [GoogleContainerTools/skaffold#5363]
`skaffold dev`

### Actual behavior
>panic: runtime error: invalid memory address or nil pointer dereference
>[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x19969cd]

### Information

- Skaffold version: 1.19.0
- Operating system: tried with MacOS, Linux
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta10
kind: Config
metadata:
  name: bug
portForward:
 -
```

**Comment by nkubala** [comment]
@msonnleitner thanks for the issue, I verified your repro and this is definitely a bug

**Comment by briandealwis** [comment]
It doesn't show up as an error when validating against our JSON schema definitions:
  - our [PortForwardResource](https://github.com/GoogleContainerTools/skaffold/blob/master/docs/content/en/schemas/v2beta13.json#L2370) doesn't mark any fields as being required. So using `portForward: [{}]` is ok too.
  - our object definitions don't mark themselves as `"type":"object"` and so allow `null`, or integers, or … (opened #5421)

It feels like we should be validating our yamls against the json schemas — I opened #5420.  I whipped up a simple test using qri-io/jsonschema but it's basically a pass-fail with no indication as to where the error is.
